packages/database/src/migrations/20230119053753-ConvertInternalDataTablesToTheirOwnTypes-modifies-schema.js

```diff
@@ -0,0 +1,44 @@
+'use strict';
+
+var dbm;
+var type;
+var seed;
+
+/**
+ * We receive the dbmigrate dependency from dbmigrate initially.
+ * This enables us to not have to rely on NODE_PATH.
+ */
+exports.setup = function (options, seedLink) {
+  dbm = options.dbmigrate;
+  type = dbm.dataType;
+  seed = seedLink;
+};
+
+exports.up = async function (db) {
+  // Copy sync_cursor over to data_service_sync_group
+  const internalDataTables = (await db.runSql("SELECT * FROM data_table WHERE type = 'internal'"))
+    .rows;
+  for (let i = 0; i < internalDataTables; i++) {
+    const dataTable = internalDataTables[i];
+    await db.runSql(
```

Comment:
These weren't working for me 😢 

Enum juggling is weird, this worked for me:
```
exports.up = async function () {
  const db = new TupaiaDatabase();

  const internalDataTables = await db.executeSql(
    "SELECT * FROM data_table WHERE type = 'internal'",
  );

  const newEnum = internalDataTables.map(dt => dt.code);
  await db.executeSql(
    `
      ALTER TYPE data_table_type RENAME TO data_table_type_temp$;
      CREATE TYPE data_table_type AS ENUM(${arrayToDbString(newEnum)});
      ALTER TABLE data_table DROP COLUMN type;
      ALTER TABLE data_table ADD COLUMN type data_table_type;
      DROP TYPE data_table_type_temp$;
    `,
    [],
  );

  for (let i = 0; i < internalDataTables.length; i++) {
    const dataTable = internalDataTables[i];
    await db.executeSql(`UPDATE data_table SET type = ? WHERE code = ?;`, [
      dataTable.code,
      dataTable.code,
    ]);
  }
};
```

Reply:
And then re-add sql into the enum...

Reply:
Ah thanks so much for this! 🙏  

My local db isn't running migrations atm and getting the latest dump is taking some time. So I've just been blindly writing SQL which of course is a reliable method of development 😉